File: tcb_project_customization/public/js/custom_direct_timesheet.js
frappe.ui.form.on("Direct Timesheet",{
    refresh:function(frm){
        if(frm.doc.docstatus==1){
            frm.add_custom_button("DPR Detail",()=>{
                frappe.new_doc("DPR Detail")
            })
        }
    }
})

File: tcb_project_customization/doc_events/custom_direct_timesheet.py
import frappe
from frappe.utils import nowdate

def validate(doc,method=None):
    today_dprs = frappe.db.get_all("Direct Timesheet",filters = {"start_date":doc.start_date,"project":doc.project},fields=["name","project","project_name"])
    if len(today_dprs)>0    :
        for entry in today_dprs:
            frappe.throw(msg = "Existing DPR exists for <b>{0}</b> for this date - <b>{1}</b>".format(entry.project_name,entry.name))